src/module_event_handlers.c

```diff
@@ -144,6 +147,7 @@ static void _FlushDBHandler(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t
 							void *data) {
 	if(eid.id == REDISMODULE_EVENT_FLUSHDB && subevent == REDISMODULE_SUBEVENT_FLUSHDB_START) {
 		// If a flushall occurs during replication, stop all decoding.
+		aux_field_counter = 0;
```

Comment:
Is it possible to receive a `flushall` then receive an `AfterKeyspaceEvent`? If so, `aux_field_counter` would wraparound to UINT_MAX.

Reply:
If a user do a `flushall` during replicating it is basically an erroneous user behavior. It is a global replica-of problem which I don't think we should handle in the module. This is an approach I also talked with Oran about and this is totaly fine.